### PR TITLE
Fix definition of length in docstring and remove redundant docstring

### PR DESCRIPTION
**Pull request by nalimilan**
`length()` is not the correct way to get the last index e.g. for strings.
Remove the docstring for AbstractArray, which is now equivalent (and
its examples, which are redundant).

[av skip]

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -118,18 +118,6 @@ ndims(::AbstractArray{T,N}) where {T,N} = N
 ndims(::Type{AbstractArray{T,N}}) where {T,N} = N
 ndims(::Type{T}) where {T<:AbstractArray} = ndims(supertype(T))
 
-"""
-    length(A::AbstractArray) -> Integer
-
-Returns the number of elements in `A`.
-
-```jldoctest
-julia> A = ones(3,4,5);
```
Comment (tkelman) [inline base/abstractarray.jl]: a multidimensional example/doctest would be good to keep

Comment (nalimilan) [inline base/abstractarray.jl]: Right, the `;` confused me (I thought that was a matrix). I've added a new one and adapted the existing one so that should be even more explicit.

**Review by ararslan — APPROVED** [review]

File: base/docs/helpdb/Base.jl [hunk 2]
```diff
@@ -1767,16 +1767,19 @@ deserialize
 """
     length(collection) -> Integer
 
-For ordered, indexable collections, returns the maximum index `i` for which `getindex(collection, i)`
-is valid.
-For unordered collections, returns the number of elements.
+Return the number of elements in collection.
```
Comment (fredrikekre) [inline base/docs/helpdb/Base.jl]: ``
`collection`
``
?

Actually, maybe change to `length(A) -> Integer` since the word collection is used to reference something else then the argument just below?

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: Well, that's how it's done elsewhere, e.g. for `endof`. The "collection" mentioned in this docstring can always be interpreted as "the argument `collection`". `A` evokes an array rather than a general collection.

Though we could certainly change our conventions, but I'd rather do this in a separate PR.

Comment (fredrikekre) [inline base/docs/helpdb/Base.jl]: > The "collection" mentioned in this docstring can always be interpreted as "the argument `collection`"

Arguments are always code highlighted, are they not?

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: Yes, here it's not really the argument, it's just the English word, but if you understand it as the argument it also works, so I don't think there's a problem.

Comment (fredrikekre) [inline base/docs/helpdb/Base.jl]: I'm confused. If its not referencing the argument, should it be `a collection` or `the collection` then (its `a collection` in the sentence below). Is it not better to disambiguate like this:

```julia
"""
    length(A) -> Integer

Return the number of elements in [collection] `A`.

Use [`endof`](@ref) to get the last valid index of an indexable collection.
"""

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: OK, I've added "the" to match the docstring for `endof`.

**Review by Sacha0 — APPROVED** [review]
lgtm! :)

**Review by rfourquet — APPROVED** [review]

**Comment by nalimilan** [comment]
Timeout on 32-bit Linux is unrelated.
